File: admin-api/src/mujahid_admin/services/status.py
from __future__ import annotations

import time

from mujahid_admin.clients.bot import bot_client
from mujahid_admin.core.config import settings
from mujahid_admin.db.connection import ping_mongo
from mujahid_admin.schemas.status import StatusResponse

_started_at = time.time()


async def get_status() -> StatusResponse:
    bot_ok = await bot_client.health_check()
    mongo_ok = await ping_mongo()
    return StatusResponse(
        status="UP" if bot_ok and mongo_ok else "DEGRADED",
        uptime_seconds=int(time.time() - _started_at),
        bot_reachable=bot_ok,
        mongo_reachable=mongo_ok,
        environment=settings.environment,
    )

File: admin-api/src/services/status.ts
import { botClient } from "@/clients/bot";
import { settings } from "@/core/config";
import { pingMongo } from "@/db/connection";
import type { StatusResponse } from "@/types";

const startedAt = Date.now();

export async function getStatus(): Promise<StatusResponse> {
  const [botOk, mongoOk] = await Promise.all([botClient.healthCheck(), pingMongo()]);
  return {
    status: botOk && mongoOk ? "UP" : "DEGRADED",
    uptimeSeconds: Math.floor((Date.now() - startedAt) / 1000),
    botReachable: botOk,
    mongoReachable: mongoOk,
    environment: settings.environment,
  };
}
